### PR TITLE
UNG-1756 deadlock

### PR DESCRIPTION
**Pull request by leroxyl**
**Fixed:**

- do not let chainer return before server is shut down
- do not accept any more requests if buffer is full

### CODE REVIEW

File: main/main.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func main() {
 		protocol:       &p,
 		env:            conf.Env,
 		authServiceURL: conf.Niomon,
-		MessageHandler: make(chan HTTPRequest, 100),
+		MessageHandler: make(chan HTTPRequest, 200),
```
Comment (ailox) [inline main/main.go]: Why 200? Can we make this configurable?

File: main/http_server.go [hunk 2]
```diff
@@ -30,9 +30,9 @@ const (
 	JSONType     = "application/json"
 	HashLen      = 32
 
-	GatewayTimeout        = 60 * time.Second
-	ShutdownTimeout       = 30 * time.Second
-	BackendRequestTimeout = 30 * time.Second
+	GatewayTimeout        = 55 * time.Second
+	ShutdownTimeout       = 60 * time.Second
+	BackendRequestTimeout = 20 * time.Second
```
Comment (ailox) [inline main/http_server.go]: Can you add a small comment how you arrived at this Timeouts?

File: main/signer.go [hunk 3]
```diff
@@ -56,35 +55,31 @@ type Signer struct {
...
-	for {
-		select {
-		case msg := <-s.MessageHandler:
-			// buffer last previous signature to be able to reset it in case sending UPP to backend fails
-			prevSign, found := s.protocol.Signatures[msg.ID]
-			if !found {
-				prevSign = make([]byte, 64)
-			}
+func (s *Signer) chainer() error {
+	for msg := range s.MessageHandler {
```
Comment (ailox) [inline main/signer.go:59]: I like the range, simplifies the handling

**Review by ailox — CHANGES_REQUESTED** [review]
Looks good, i needed some time to understand the code. I think some comments for the numbers would help, e.g. "55s is when we can expect the connection to niomon to be lost", "200 messages is enough to provide a buffer for 50 devices", or similar.

File: main/main.go (hunk 1 above)
Comment (leroxyl) [reply main/main.go]: i added a comment and will note to make it configurable in a future release

File: main/http_server.go (hunk 2 above)
Comment (leroxyl) [reply main/http_server.go]: done

File: main/signer.go (hunk 3 above)
Comment (leroxyl) [reply main/signer.go:59]: :)

**Review by ailox — APPROVED** [review]

**Pull request merged**